common/djangoapps/track/backends/tests/test_adaptive_learning.py

```diff
@@ -0,0 +1,154 @@
+from __future__ import absolute_import
+
+import ddt
+from django.test import TestCase
+from mock import Mock, patch
+
+from track.backends.adaptive_learning import (
+    _is_problem_check, _get_course, _get_block_id, _get_user_id, _get_success, AdaptiveLearningBackend
+)
+
+
+@ddt.ddt
+class TestAdaptiveLearningBackend(TestCase):
+    """
+    Tests for AdaptiveLearningBackend and helper functions.
+    """
+
+    def setUp(self):
+        super(TestAdaptiveLearningBackend, self).setUp()
+        self.backend = AdaptiveLearningBackend()
+
+    @ddt.data(
+        ('problem_check', True),
+        ('other', False)
+    )
+    @ddt.unpack
+    def test__is_problem_check(self, event_type, expected_result):
+        """
+        Test that `_is_problem_check` function returns True if event is of type "problem_check",
+        and False otherwise.
+        """
+        event = {'event_type': event_type}
+        result = _is_problem_check(event)
+        self.assertEqual(result, expected_result)
+
+    def test__get_course(self):
+        """
+        Test that `_get_course` function extracts appropriate value from event and returns it.
+        """
+        course_id = 'block-v1:org+course+run+type@course+block@course'
+        event = {
+            'context': {
+                'course_id': course_id
+            }
+        }
+        with patch('track.backends.adaptive_learning.CourseKey') as patched_class, \
+             patch('track.backends.adaptive_learning.modulestore') as patched_modulestore:
+            mock_course_key = Mock()
+            patched_class.from_string.return_value = mock_course_key
+            mock_modulestore = Mock()
+            mock_course = Mock()
```

Comment:
@itsjeyd now you see why `modulestore()` is a bad idea? :) Suggestion: make modulestore passed into `AdaptiveLearningBackend` (as a constructor parameter?), and just pass a mocked instance to `AdaptiveLearningBackend` instantiation.

Oh, yes, and this requires refactoring all those methods into static/class/instancemethods and redoing the test suite into testing `AdaptiveLearningBackend`. Which is something you should do anyway :)